File: be/src/labels/dto/create-label.dto.ts
import { IsNotEmpty, IsString, IsOptional } from "class-validator";

export class CreateLabelDto {
  @IsString()
  @IsOptional()
  name?: string;

  @IsString()
  @IsNotEmpty()
  color: string;

  @IsString()
  @IsNotEmpty()
  projectId: string;
}

File: backend-java/src/main/java/com/qlda/backend/labels/dto/CreateLabelDto.java
package com.qlda.backend.labels.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class CreateLabelDto {
    private String name;

    @NotBlank(message = "Color không được để trống")
    private String color;

    @NotBlank(message = "Project ID không được để trống")
    private String projectId;
}

